tc-client/src/main/java/com/tc/object/CloseHookCallable.java

```diff
@@ -0,0 +1,30 @@
+/*
+ *
+ *  The contents of this file are subject to the Terracotta Public License Version
+ *  2.0 (the "License"); You may not use this file except in compliance with the
+ *  License. You may obtain a copy of the License at
+ *
+ *  http://terracotta.org/legal/terracotta-public-license.
+ *
+ *  Software distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ *  the specific language governing rights and limitations under the License.
+ *
+ *  The Covered Software is Terracotta Core.
+ *
+ *  The Initial Developer of the Covered Software is
+ *  Terracotta, Inc., a Software AG company
+ *
+ */
+package com.tc.object;
+
+import java.util.concurrent.Callable;
+import org.terracotta.exception.EntityException;
+
+/**
+ *
+ */
+public interface CloseHookCallable extends Callable<EntityException> {
```

Comment:
Do we need an interface here? What's wrong with just using the generic declaration directly?

Reply:
The exception handling is messy.  Runnable doesn't really fit and I wanted a Callable that didn't throw.

Reply:
We are just suppressing and/or logging these errors so I can move it to a generic callable